### PR TITLE
Set user.principal in a way spring security can be covered

### PR DESCRIPTION
**Pull request by tylerbenson**
Improve tests.

### CODE REVIEW

File: dd-java-agent/instrumentation/servlet-2/src/main/java/datadog/trace/instrumentation/servlet2/Servlet2Advice.java [hunk 1]
```diff
@@ -49,16 +50,24 @@ public static Scope startSpan(
...
+      @Advice.Enter final Scope scope,
+      @Advice.Argument(0) final ServletRequest req,
+      @Advice.Thrown final Throwable throwable) {
+    // Set user.principal regardless of who created this span.
+    final Span currentSpan = GlobalTracer.get().activeSpan();
+    if (currentSpan != null) {
+      if (req instanceof HttpServletRequest) {
+        final Principal principal = ((HttpServletRequest) req).getUserPrincipal();
+        if (principal != null) {
+          currentSpan.setTag("user.principal", principal.getName());
```
Comment (mar-kolya) [inline dd-java-agent/instrumentation/servlet-2/src/main/java/datadog/trace/instrumentation/servlet2/Servlet2Advice.java]: Could you please add 'standard' tag name constant for this - it is used in multiple places.

File: dd-java-agent/instrumentation/spring-web/spring-web.gradle [hunk 2]
```diff
@@ -32,8 +34,7 @@ dependencies {
 
   testCompile group: 'org.spockframework', name: 'spock-spring', version: '1.1-groovy-2.4'
 
-  testCompile group: 'org.springframework', name: 'spring-web', version: '4.3.14.RELEASE'
-  testCompile group: 'org.springframework', name: 'spring-webmvc', version: '4.3.14.RELEASE'
-  testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: '1.5.10.RELEASE'
-  testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-tomcat', version: '1.5.10.RELEASE'
+  testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: '1.5.+'
```
Comment (mar-kolya) [inline dd-java-agent/instrumentation/spring-web/spring-web.gradle]: Would it make sense to specify exact version here instead to make sure that our build is repeatable? I think we do this for dependencies in other instrumentations.

File: dd-java-agent/instrumentation/servlet-2/src/main/java/datadog/trace/instrumentation/servlet2/Servlet2Advice.java (hunk 1 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/servlet-2/src/main/java/datadog/trace/instrumentation/servlet2/Servlet2Advice.java]: done.

File: dd-java-agent/instrumentation/spring-web/spring-web.gradle (hunk 2 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/spring-web/spring-web.gradle]: done.

**Pull request merged**